spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/DataSet.scala

```diff
@@ -280,7 +282,12 @@ class CachedDistriDataSet[T: ClassTag] private[dataset]
 
         override def next(): T = {
           val i = _offset.getAndIncrement()
+
           if (_train) {
+            if (localData.length == 0) {
+              throw DatatoosmallException("dataset to small, please make increase dataset size to" +
+                " ensure no mapping localData to size zero")
+            }
```

Comment:
require(localData.length != 0, "dataset on this executor is empty, please increase your dataset size or decrease your number of executors.")